openstack/02a-MAClow/IEEE802154E.c

```diff
@@ -1536,21 +1582,36 @@ port_INLINE void activity_ri6() {
                                      IEEE802154E_DESC_TYPE_SHORT; 
    ieee154e_vars.ackToSend->payload[0] = ((header_desc.length_elementid_type) >> 8) & 0xFF;
    ieee154e_vars.ackToSend->payload[1] = (header_desc.length_elementid_type)        & 0xFF;
-   
+
    // prepend the IEEE802.15.4 header to the ACK
    ieee154e_vars.ackToSend->l2_frameType = IEEE154_TYPE_ACK;
    ieee154e_vars.ackToSend->l2_dsn       = ieee154e_vars.dataReceived->l2_dsn;
+
+   // To send ACK, we use the same security level (including NOSEC) and keys
+   // that were present in the DATA packet.
+   ieee154e_vars.ackToSend->l2_securityLevel = ieee154e_vars.dataReceived->l2_securityLevel;
+   ieee154e_vars.ackToSend->l2_keyIdMode     = ieee154e_vars.dataReceived->l2_keyIdMode;
+   ieee154e_vars.ackToSend->l2_keyIndex      = ieee154e_vars.dataReceived->l2_keyIndex;
+
    ieee802154_prependHeader(ieee154e_vars.ackToSend,
                             ieee154e_vars.ackToSend->l2_frameType,
                             TRUE,//ie in ack
-                            IEEE154_SEC_NO_SECURITY,
+                            ieee154e_vars.ackToSend->l2_securityLevel == IEEE154_ASH_SLF_TYPE_NOSEC ? 0 : 1,

```

Comment:
Because it expects bool, not the security level (this corresponds to the security enabled bit in the FCF, not the security level in the security header). We can cast ```ieee154e_vars.ackToSend->l2_securityLevel``` to bool if you think it's more clear?